kpops/utils/environment.py

```diff
@@ -0,0 +1,36 @@
+import os
+import platform
+from collections import UserDict
+from typing import Callable
+
+
+class Environment(UserDict):
+    def __init__(self, mapping=None, /, **kwargs) -> None:
+
+        transformation = self.__get_transformation()
+        if mapping is not None:
+            mapping = {transformation(key): value for key, value in mapping.items()}
+        else:
+            mapping = {}
+        if kwargs:
+            mapping.update(
+                {transformation(key): value for key, value in kwargs.items()}
+            )
+        super().__init__(mapping)
+
+    @staticmethod
+    def __key_camel_case_transform(key: str) -> str:
+        return str(key).lower()
+
+    @staticmethod
+    def __key_identity_transform(key: str) -> str:
+        return str(key)
+
+    def __get_transformation(self) -> Callable[[str], str]:
+        if platform.system() == "Windows":
+            return Environment.__key_camel_case_transform
+        else:
+            return Environment.__key_identity_transform
+
+
+environ = Environment(os.environ)
```

Comment:
what do you think about `ENV`? I was never a fan of the environ name in the os module 😄